factcast-factus/src/main/java/org/factcast/factus/FactusImpl.java

```diff
@@ -298,7 +301,14 @@ void createSnapshot(SnapshotProjection projection, UUID state) {
     SnapshotSerializer ser = snapFactory.retrieveSerializer(aggregateClass);
 
     Optional<Snapshot> latest = aggregateSnapshotRepository.findLatest(aggregateClass, aggregateId);
-    Optional<A> optionalA = latest.map(as -> ser.deserialize(aggregateClass, as.bytes()));
+    Optional<A> optionalA =
+        latest
+            .map(as -> ser.deserialize(aggregateClass, as.bytes()))
+            .map(
+                s -> {
+                  s.onAfterRestore();
+                  return s;
+                });
```

Comment:
This is annoying, I saw the suggestion to have a utility method to simulate the behavior of `Stream.peek()`:
```
<T> UnaryOperator<T> peek(Consumer<T> c) {
    return x -> {
        c.accept(x);
        return x;
    };
}
```
Then we can write in such cases `...map(peek(s -> s.onAfterRestore()))....`

Reply:
will do